File: Q338_CountingBits.py
class Solution:
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        res = []
        for i in range(num+1): #range列举的区间为左闭右开
            res.append(bin(i).count('1')) #注意区分bin和bytes两个函数的区别
        return res
        
'''
RunTime:56ms
'''

File: Q338_CountingBits.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> res(num+1, 0); //初始化可用长度为num的矢量，并初始化每个元素为0
        for(int i=1; i<=num; i++){
               res[i] = res[i & (i-1)] + 1; //骚操作：与其前边的一个数按位相与就得到其前一个数的二进制中1的个数
        }
        return res;
        
    }
};
/*
RunTime:2ms
*/